include/deal.II/fe/fe_interface.h

```diff
@@ -0,0 +1,787 @@
+      }
+    else
+      {
+        internal_fe_subface_values_neighbor.reinit(cell_neighbor,
+                                                   face_no_neighbor,
+                                                   sub_face_no_neighbor);
+        fe_face_values_neighbor = &internal_fe_subface_values_neighbor;
+      }
+
+    std::vector<types::global_dof_index> v(n_dofs_fe);
+    cell->get_dof_indices(v);
+    std::vector<types::global_dof_index> v2(n_dofs_fe_neighbor);
+    cell_neighbor->get_dof_indices(v2);
+
+    interface_dof_indices.clear();
+    interface_dof_indices.insert(interface_dof_indices.end(),
+                                 v.begin(),
+                                 v.end());
+    interface_dof_indices.insert(interface_dof_indices.end(),
+                                 v2.begin(),
+                                 v2.end());
+  }
+
+  /**
+   * Re-initialize this object to be used on a interface given by a single face
+   * of a cell. This is useful to use FEInterfaceValues to work on boundaries of
+   * the domain.
+   *
+   * As a consequence, queries like jump() will assume a value of zero for the
+   * values on the "other" side.
+   */
+  template <class Iterator>
+  void
+  reinit(const Iterator &cell, const unsigned int &face_no)
+  {
+    internal_fe_face_values.reinit(cell, face_no);
+    fe_face_values          = &internal_fe_face_values;
+    fe_face_values_neighbor = nullptr;
+
+    interface_dof_indices.resize(n_dofs_fe);
+    cell->get_dof_indices(interface_dof_indices);
+  }
+
+  /**
+   * Return the vector of JxW values for each quadrature point.
+   */
+  const std::vector<double> &
+  get_JxW_values() const
+  {
+    return internal_fe_face_values.get_JxW_values();
+  }
+
+  /**
+   * Return the normal vector of the facet in each quadrature point.
+   */
+  const std::vector<Tensor<1, spacedim>> &
+  get_normal_vectors() const
+  {
+    return internal_fe_face_values.get_normal_vectors();
+  }
+
+  /**
+   * Return a reference to the quadrature object in use.
+   */
+  const Quadrature<dim - 1> &
+  get_quadrature() const
+  {
+    return internal_fe_face_values.get_quadrature();
+  }
+
+  /**
+   * Return a reference to the quadrature object in use.
+   */
+  const std::vector<Point<spacedim>> &
+  get_quadrature_points() const
+  {
+    return internal_fe_face_values.get_quadrature_points();
+  }
+
+  /**
+   * Return the update flags set.
+   */
+  const UpdateFlags
+  get_update_flags() const
+  {
+    return internal_fe_face_values.get_update_flags();
+  }
+
+
+  /**
+   * Return the number of DoFs on the current facet. This is the sum of the
+   * number of DoFs on the two adjacent cells.
+   */
+  unsigned
+  n_interface_dofs() const
+  {
+    return n_dofs_fe + (is_boundary_facet() ? 0 : n_dofs_fe_neighbor);
+  }
+
+  /**
+   * Return if the current facet, which is set by reinit(), is an internal face
+   * with two adjacent cells or a boundary face.
+   */
+  bool
+  is_boundary_facet() const
```

Comment:
I would rename this as `at_boundary()`, similarly to what we have for `face(f)->at_boundary()`